draft-ietf-protected-audience-key-value-server-api.md

```diff
@@ -1,78 +1,175 @@
+media types need to be defined:
+
+-   The OHTTP request media type is `message/ad-auction-trusted-signals-request`
+-   The OHTTP response media type is `message/ad-auction-trusted-signals-response`
+
+These media types are concatenated with other fields when creating the [HPKE] encryption context.
+
+### Message Framing {#framing}
+
+Inside the ciphertext, the requests and responses have the following framing:
+
+| Byte     | 0         | 0             | 1 to 4   | 5 to Size+4       | Size+5 to end   |
+| -------- | --------- | ------------- | -------- | ----------------- | --------------- |
+| Bits     | 7-2       | 1-0           | \*       | \*                | \*              |
+| -------- | --------- | ------------- | -------- | ----------------- | --------------- |
+| Contents | Unused    | Compression   | Size     | Request Payload   | Padding         |
+
+The request/response is framed with a 5 byte header.
+
+The first byte is the format+compression byte. The lower 2 bits are used for compression
+specification. The higher 6 bits are currently unused.
+
+The following 4 bytes are the length of the request message in network byte order.
+
+Then the request is zero padded to a set of pre-configured lengths.
+
+### Format+compression byte {#compression}
+
+| Compression | Description                    |
+| :---------: | :----------------------------- |
+|    0x00     | [CBOR], no compression         |
+|    0x01     | [CBOR], compressed in [Brotli] |
+|    0x02     | [CBOR], compressed in [GZIP]   |
+
+For requests, the byte value is 0x00. For responses, the byte value depends on the
+`acceptCompression` field in the request and the server behavior.
+
+### Padding {#padding}
+
+Padding is applied with sizes as multiples of 2^n KBs ranging from 0 to 2MB. So the valid response
+sizes will be `[0, 128B, 256B, 512B, 1KB, 2KB, 4KB, 8KB, 16KB, 32KB, 64KB, 128KB, 256KB, 512KB, 1MB,
+2MB]`.
+
+## Core Request Data {#request}
+
+Requests are not compressed and have a tree-like hierarchy:
 
+-   Each request contains one or more partitions. Each partition is a collection of keys that can be
+    processed together by the service without any potential privacy leakage. Keys from one interest
+    group must be in the same partition. Keys from different interest groups with the same joining
+    site may or may not be in the same partition.
+-   Each partition contains one or more key groups. Each key group has its unique attributes among
+    all key groups in the partition. The attributes are represented by a list of `tags`. Besides
+    tags, the key group contains a list of keys to look up.
+-   Each partition has a unique id.
+-   Each partition has a compression group field. Results of partitions belonging to the same
+    compression group can be compressed together in the response. Different compression groups must
+    be compressed separately. See more details below.
```

Comment:
Can we link "See more details below" to the details?

Reply:
Done.